error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ImplicitBlockingFluxOperation.java

```diff
@@ -0,0 +1,140 @@
+import com.sun.tools.javac.util.Position;
+import java.util.Optional;
+import tech.picnic.errorprone.bugpatterns.util.SourceCode;
+import tech.picnic.errorprone.bugpatterns.util.ThirdPartyLibrary;
+
+/**
+ * A {@link BugChecker} that flags usages of {@link reactor.core.publisher.Flux} methods that are
+ * documented to block, but that behaviour is not apparent from their signature.
+ *
+ * <p>The alternatives suggested are explicitly blocking operators, highlighting actual behavior,
+ * not adequate for automatic replacements, they need manual review.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary = "Accidental blocking of `Flux` with convenience method",
+    explanation =
+        "`Flux#toStream` and `Flux#toIterable` are documented to block, "
+            + "but this is not apparent from the method signature; "
+            + "please make sure that they are used with this in mind",
+    link = BUG_PATTERNS_BASE_URL + "ImplicitBlockingFluxOperation",
+    linkType = CUSTOM,
+    severity = SUGGESTION,
+    tags = STYLE)
+public final class ImplicitBlockingFluxOperation extends BugChecker
+    implements MethodInvocationTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Matcher<ExpressionTree> FLUX_WITH_IMPLICIT_BLOCK =
+      instanceMethod()
+          .onDescendantOf(Suppliers.typeFromString("reactor.core.publisher.Flux"))
+          .namedAnyOf("toIterable", "toStream")
+          .withNoParameters();
+  private static final Supplier<Type> STREAM = Suppliers.typeFromString("java.util.stream.Stream");
+
+  /** Instantiates a new {@link ImplicitBlockingFluxOperation} instance. */
+  public ImplicitBlockingFluxOperation() {}
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    if (!FLUX_WITH_IMPLICIT_BLOCK.matches(tree, state)) {
+      return Description.NO_MATCH;
+    }
+
+    Description.Builder description = buildDescription(tree);
+
+    description.addFix(SuggestedFixes.addSuppressWarnings(state, "ImplicitBlockingFluxOperation"));
+    if (ThirdPartyLibrary.GUAVA.isIntroductionAllowed(state)) {
+      trySuggestFix("com.google.common.collect.ImmutableList.toImmutableList", tree, state)
+          .ifPresent(description::addFix);
+    }
+    trySuggestFix("java.util.stream.Collectors.toUnmodifiableList", tree, state)
+        .ifPresent(description::addFix);
+
+    return description.build();
+  }
+
+  private static Optional<SuggestedFix> trySuggestFix(
+      String fullyQualifiedMethodInvocation, MethodInvocationTree tree, VisitorState state) {
+    SuggestedFix.Builder fix = SuggestedFix.builder();
+    String replacement =
+        SuggestedFixes.qualifyStaticImport(fullyQualifiedMethodInvocation, fix, state);
+
+    return replaceMethodInvocationWithCollect(tree, replacement + "()", fix.build(), state);
+  }
+
+  // XXX: Assumes that the generated `collect(...)` expression will evaluate to
```

Comment:
This does make sense because we convert the method. The resulting type of the complete expression will not change, so the behavior is "preserved". So will drop the `XXX` 😄.